app/src/services/AdService.ts

```diff
@@ -30,4 +45,138 @@ export default class AdService {
       };
     }
   }
+
+  /**
+   * This methods get all the ads from the connected firebase
+   * @param book : Book (optional) will find ads with this book
+   * @param user : User (optional) will find ads with this user
+   * @returns Promise<ServiceSuccessResponse>
+   */
+  private static async getAds(
+    book?: Book | null,
+    user?: User | null,
+  ): Promise<ServiceSuccessResponse> {
+    const queryConstraints = [];
+    if (book) queryConstraints.push(where('bookId', '==', book.uid));
+    if (user) queryConstraints.push(where('uid', '==', user.uid));
+
+    const q = query(collection(db, 'ads'), ...queryConstraints);
+
+    const ads: Ad[] = [];
+    const querySnapshot = await getDocs(q);
+
+    querySnapshot.forEach((docs) => {
+      const ad: Ad = {
+        uid: docs.id,
+        userId: docs.data().uid,
+        bookId: docs.data().bookId,
+        price: docs.data().price,
+        condition: docs.data().condition,
+        conditionDescribtion: docs.data().conditionDescribtion,
+      };
+
+      ads.push(ad);
+    });
+
+    return { success: true, data: ads };
+  }
+
+  /**
+   * This edits a document in the connected Firebase Firestore
+   * @param ad the ad that should be edited
+   * @param data the data that should be changed, Ex input {condition: newCondition} or {price: newPrice}
+   * @returns Promise<ServiceSuccessResponse>
+   */
+  private static async editAd(
```

Comment:
I think the UID should be enough to edit an ad, it doesn't make sense that you would need an entire ad object

Reply:
Fixed